packages/postgres/src/pool.ts

```diff
@@ -254,40 +229,37 @@ class PostgresPool {
     // It does NOT support clearing the existing search schema - e.g.,
     // `setSearchSchema(null)` would not work as you expect. This is fine, as
     // that's not something we ever do in practice.
-    if (this.searchSchema != null && client[SEARCH_SCHEMA] !== this.searchSchema) {
+    if (this.searchSchema != null && (client as any)[SEARCH_SCHEMA] !== this.searchSchema) {
```

Comment:
Can we say it's a `pg.PoolClient`?

Reply:
This `any` cast is necessary because we try to access a property (`SEARCH_SCHEMA`) that isn't declared in the types.

If we wanted to avoid this, we could probably instead maintain a `WeakMap` from client to the search schema that it's using?

Reply:
Oh right, we've (probably me) done a horrible thing here. I agree the clean thing (for TS) is to use a WeakMap. I'm ok if you don't want to make that change in this PR.

Reply:
I went ahead and made this change in 5ae6ac2. I'd appreciate one final look to make sure I didn't mess anything up!